xmppserver/src/main/java/org/jivesoftware/openfire/Connection.java

```diff
@@ -368,6 +382,33 @@ public interface Connection extends Closeable {
      */
     ConnectionConfiguration getConfiguration();
 
+    /**
+     * When a connection is used to transmit an XML data, the root element of that data can define XML namespaces other
+     * than the ones that are default (eg: 'jabber:client', 'jabber:server', etc). For an XML parser to be able to parse
+     * stanzas or other elements that are defined in that namespace (eg: are prefixed), these namespaces are recorded
+     * here.
+     *
+     * @return A collection that contains all non-default namespaces that the peer defined when last opening a new stream.
+     */
+    @Nonnull
+    default Set<Namespace> getAdditionalNamespaces() {
+        return additionalNamespaces;
```

Comment:
A possible loss of encapsulation here. Perhaps return an unmodifiable copy of the set?

Reply:
This will be invoked for _every_ stanza that is received over the connection, while the added benefit is low. I'm not sure if the added overhead is worth the safety that it adds.